graphql/query_resolver.go

```diff
@@ -618,11 +618,17 @@ func (r *queryResolver) TaskLogs(ctx context.Context, taskID string, execution *
 func (r *queryResolver) TaskTests(ctx context.Context, taskID string, execution *int, sortCategory *TestSortCategory, sortDirection *SortDirection, page *int, limit *int, testName *string, statuses []string, groupID *string) (*TaskTestResult, error) {
 	dbTask, err := task.FindByIdExecution(taskID, execution)
 	if dbTask == nil || err != nil {
-		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("finding task with id %s", taskID))
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("finding task with id '%s'", taskID))
+	}
+
+	var baseTask *task.Task
+	if evergreen.IsPatchRequester(dbTask.Requester) {
+		baseTask, err = dbTask.FindTaskOnBaseCommit()
+	} else {
+		baseTask, err = dbTask.FindTaskOnPreviousCommit()
```

Comment:
I don't know what is the current/desired behavior here, so I'm just asking to make sure - Will this do the correct thing for non-commit tasks on the waterfall, like periodic builds and downstream builds? For example, periodic builds are usually a re-run of the latest commit.

Reply:
I ended up swapping the conditional to only look for previous commit if the requester is `"gitter_request"`!

From my understanding, patch requesters, periodic build requesters, git tag requesters, and trigger requesters should look back at their base commit based on the `revision` rather than the `order` number. I added more tests to verify this works as intended.

Let me know if that sounds right!

Reply:
This sounds good to me! 👍